stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt

```diff
@@ -283,24 +279,21 @@ internal class ChatDomainImpl internal constructor(
         logger.logI("Initializing ChatDomain with version " + getVersion())
 
         // if the user is already defined, just call setUser ourselves
-        val current = userOverwrite ?: client.getCurrentUser()
```

Comment:
How do the changes in this class not break existing behaviour for users who have been using the three-param constructor, passing in a `User` object to the builder? We were previously performing logic based on that parameter, and now we're ignoring it.

Reply:
Even if they passed user they could do nothing unless they invoke `ChatClient::connectUser`. IMO it was useless to have such logic from the moment when we had brought these listeners for set/unset users in ChatClient, wasn't it?

Reply:
Also, it was used for such style, when you set/unset user you should create a new instance of ChatDomain. It can work with these changes as well

Reply:
I'm honestly not quite sure what behaviour changes (if any) removing this will do, but it'd be easier and safer to just leave this code in place, and remove it later when we remove the public-facing API.